### PR TITLE
Remove legacy schema

### PR DESCRIPTION
**Pull request by wizardofoz123**
This is an attempt to remove the old (legacy) db schema from kpet.

### CODE REVIEW

File: kpet/schema.py [hunk 1]
```diff
@@ -158,43 +158,6 @@ def resolve(self, data):
...
-            # Else it's a conversion function, and if we found valid schema
-            elif last_valid_schema:
-                # Convert the data for the next schema/converter
-                data = schema_or_converter(data)
-        # We should arrive at the last schema
-        assert last_valid_schema is self.schemas_and_converters[-1]
-        # Resolve the data with the last schema
-        return last_valid_schema.resolve(data)
-
-
```
Comment (spbnick) [inline kpet/schema.py]: Please don't remove the Succession schema implementation, we'll keep using it :smile:!

File: tests/assets/db/invalid/semantics/missing_case/suite.yaml [hunk 2]
```diff
@@ -1,6 +1,3 @@
 description: suite1
 cases:
     - name: case1
-patterns:
-    - pattern: a
-      case_name: case2
```
Comment (spbnick) [inline tests/assets/db/invalid/semantics/missing_case/suite.yaml:6]: We need to remove this database and the associated test case altogether as they no longer apply. The test should have a TODO about this.

**Review by spbnick — CHANGES_REQUESTED** [review]
Looks great, thanks a lot, Jakub! Requesting a couple tweaks only.

**Comment by wizardofoz123** [comment]
@cki-bot: test this, bot!

**Review by spbnick — APPROVED** [review]
Looks good! Thank you, Jakub :)

**Comment by wizardofoz123** [comment]
Thank you too Nick, all those suggested tweaks were valuable. :)

**Pull request merged**